core/include/cubos/core/al/audio_device.hpp

```diff
@@ -51,36 +43,40 @@ namespace cubos::core::al
         /// @brief Forbid copy construction.
         AudioDevice(const AudioDevice&) = delete;
 
-        /// @brief Creates an audio device from a given device @p specifier.
+        /// @brief Creates an audio device.
         /// @see enumerateDevices()
-        /// @param specifier Device specifier (empty for default).
         /// @return Audio device, or nullptr on failure.
-        static std::shared_ptr<AudioDevice> create(const std::string& specifier = "");
-
-        /// @brief Enumerates the available devices.
-        /// @param[out] devices Vector to fill with the available devices.
-        static void enumerateDevices(std::vector<std::string>& devices);
+        static std::shared_ptr<AudioDevice> create();
 
         /// @brief Creates a new audio buffer
+        /// @param filePath File path to create buffer from.
         /// @return Handle of the new buffer.
-        virtual Buffer createBuffer() = 0;
+        static Buffer createBuffer(const void* data, size_t dataSize);
 
         /// @brief Creates a new audio source.
         /// @return Handle of the new source.
-        virtual Source createSource() = 0;
+        static Source createSource();
+
+        /// @brief Enumerates the available devices.
+        /// @param[out] devices Vector to fill with the available devices.
+        static void enumerateDevices(std::vector<std::string>& devices);
 
         /// @brief Sets the position of the listener.
         /// @param position Position.
-        virtual void setListenerPosition(const glm::vec3& position) = 0;
+        /// @param listenerIndex Index of the listener
+        virtual void setListenerPosition(const glm::vec3& position, unsigned int listenerIndex = 0) = 0;
 
         /// @brief Sets the orientation of the listener.
         /// @param forward Forward direction of the listener.
         /// @param up Up direction of the listener.
-        virtual void setListenerOrientation(const glm::vec3& forward, const glm::vec3& up) = 0;
+        /// @param listenerIndex Index of the listener
+        virtual void setListenerOrientation(const glm::vec3& forward, const glm::vec3& up,
+                                            unsigned int listenerIndex = 0) = 0;
 
         /// @brief Sets the velocity of the listener. Used to implement the doppler effect.
         /// @param velocity Velocity of the listener.
-        virtual void setListenerVelocity(const glm::vec3& velocity) = 0;
+        /// @param listenerIndex Index of the listener
+        virtual void setListenerVelocity(const glm::vec3& velocity, unsigned int listenerIndex = 0) = 0;
```

Comment:
What does it mean for there to be different listeners at once? :thinking: And if there are multiple listeners, should they maybe moved to a different `AudioListener` object, as we do with sources?